eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +119,117 @@ function Confirm-ChangeLogEntry {
   return $true
 }
 
-function Set-TestChangeLog($TestVersion, $changeLogFile, $ReleaseEntry) {
-  Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status="(Unreleased)",
+    [String]$Title,
+    [String[]]$Content
+  )
+
+  # Validate relase Status
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    $dateFormat = "yyyy-MM-dd"
+    $provider = [System.Globalization.CultureInfo]::InvariantCulture
+    try {
+      $Status = ([System.DateTime]::ParseExact($Status, $dateFormat, $provider)).ToString($dateFormat)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$dateFormat' or use '(Unreleased)'"
+    }
+  }
+
+  $Status = "($Status)"
+  if (!$Content) { $Content = @() }
+  if (!$Title) { $Title = "## $Version $Status" }
+
+  $newChangeLogEntry = [pscustomobject]@{ 
```

Comment:
Should we try to parse the Version?

Reply:
I was going to do that but I wonder if that's necessary here. We would need to determine if its a python version because there is a different parse function for that. So far the changelog logic have not been language dependent and I wanted to keep it that way.

Reply:
Maybe we can move SetupPythonConventions into the pythons Language setting, Then we can keep ParseVersionString in common.ps1 and only call SetupPythonConventions  if it has been loaded from python Language Settings file.